ycmd.el

```diff
@@ -352,6 +357,23 @@ Returns the new value of `ycmd-force-semantic-completion`.
              (if force "enabled" "disabled"))
     (setq ycmd-force-semantic-completion force)))
 
+(defun ycmd--get-tags-file (buffer)
+  "Get tags files for current BUFFER or nil.
+
+Use `tags-file-name' variable if set, or look up directory
+hierarchy for first found tags file."
+  (or tags-file-name
+      (let* ((directory (locate-dominating-file
```

Comment:
Should we do this? Going up the directory hierarchy for a possible tags file? or should we only use `tags-file-name` or should we event introduce a variable in ycmd?